todo-backend/README.adoc

```diff
@@ -0,0 +1,452 @@
+$ curl -X POST -H "Content-Type: application/json"  -d '{"title": "This is my first todo item!"}' http://localhost:8080/
+{"completed":false,"id":1,"order":0,"title":"This is my first todo item!","url":"https://localhost:8080/1"}%
+
+# get a list of todos with the one that was just created
+$ curl http://localhost:8080
+[{"completed":false,"id":1,"order":0,"title":"This is my first todo item!","url":"https://localhost:8080/1"}
+----
+endif::[]
+
+== Run the Backend on OpenShift
+
+=== Prerequisites
+
+* You must be logged in OpenShift and have an `oc` client to connect to OpenShift
+* https://helm.sh[Helm] must be installed to deploy the backend on OpenShift.
+
+Once you have installed Helm, you need to add the repository that provides Helm Charts for {productName}:
+
+ifndef::ProductRelease[]
+[source,options="nowrap"]
+----
+$ helm repo add wildfly https://docs.wildfly.org/wildfly-charts/
+"wildfly" has been added to your repositories
+$ helm search repo wildfly
+NAME                    CHART VERSION   APP VERSION     DESCRIPTION
+wildfly/wildfly         1.1.0           22.0            Build and Deploy WildFly applications on OpenShift
+----
+endif::[]
+ifdef::ProductRelease[]
+[source,options="nowrap"]
+----
+$ helm repo add jboss-eap https://jbossas.github.io/eap-charts/
+"jboss-eap" has been added to your repositories
+$ helm search repo jboss-eap
+NAME                    CHART VERSION   APP VERSION     DESCRIPTION
+jboss-eap/eap7          1.0.0           7.4             A Helm chart to build and deploy EAP 7 applic...
+----
+endif::[]
+
+
+=== Deploy a PostgreSQL Database on OpenShift
+
+[source,options="nowrap"]
+----
+$ oc new-app postgresql-ephemeral \
+   -p DATABASE_SERVICE_NAME=todos-db \
+   -p POSTGRESQL_DATABASE=todos
+----
+
+This will create a PostgreSQL database named `todos` on OpenShift that can be accessed on the port `5432` on the service `todos-db`.
+We don't need to copy the credentials to connect to the database as we will retrieve them later using the `todos-db` secret that was created
+when the database is deployed.
+
+ifndef::ProductRelease[]
+=== Build and Deploy the Backend on OpenShift with Bootable Jar
+
+The backend will be built and deployed on OpenShift with a Helm Chart for {productName}.
+
+[source,options="nowrap"]
+----
+$ helm install todo-backend -f todo-backend-bootable-jar.yaml wildfly/wildfly
+NAME: todo-backend
+...
+STATUS: deployed
+REVISION: 1
+----
+
+The Helm Chart for this quickstart contains all the information to build an image from the source code using Bootable Jar:
+
+[source,options="nowrap"]
+----
+build:
+  uri: https://github.com/wildfly/quickstart.git
+  mode: bootable-jar
+----
+endif::[]
+
+=== Build and Deploy the Backend on OpenShift with {productName} S2I
+
+The backend will be built and deployed on OpenShift with a Helm Chart for {productName}.
+
+ifndef::ProductRelease[]
+[source,options="nowrap"]
+----
+$ helm install todo-backend -f todo-backend-s2i.yaml wildfly/wildfly
```

Comment:
when user reads and tries the README from top down then this command will fail as the chart `todo-backend` is already installed via previous steps
```
Error: cannot re-use a name that is still in use
```

We can refer to the section `Remove the Backend` to clean it first

Reply:
that's fine. The user is not supposed to install both S2I and bootable jar examples at the same time.

Reply:
and if if wants to install 2 applications, he just have to specify another name